File: fixtures/conformance/ruby/deep_nesting.rb
# Anidamiento profundo (5 niveles: for>if>while>begin>if). begin/ensure es el
# try/finally de Ruby: anida SIN ser decision (analogo del `with` de Python).
def deep_nesting(items)
  for a in items
    if a
      while a
        begin
          if a
            return a
          end
        ensure
          nil
        end
      end
    end
  end
  0
end

File: fixtures/conformance/c/deep_nesting.c
/* Anidamiento profundo (5 niveles: for>if>while>bloque etiquetado>if). El bloque
   etiquetado (labeled_statement, idioma del goto de limpieza) anida SIN ser
   decision: es el analogo local del `with` de Python (C no tiene try/with/lock). */
int deep_nesting(int n) {
    for (int a = 0; a < n; a++) {
        if (a > 0) {
            while (a > 0) {
                cleanup: {
                    if (a > 0) {
                        return a;
                    }
                }
            }
        }
    }
    return 0;
}
